transform/app/tasks/delete_data_by_id.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import requests
+from requests import ConnectionError as ReqConnectionError
+
+from app.services.solr.utils import ids_mapping
+from app.settings import settings
+from app.worker import celery
+
+logger = logging.getLogger(__name__)
+
+
+@celery.task(name="delete_data_by_id")
+def delete_data_by_id(
+    col_name: str,
+    data: dict | list[dict],
+) -> None:
+    """Delete solr resource based on its ID"""
+    raw_id = data[0]["id"] if col_name == "interoperability guideline" else data["id"]
```

Comment:
it should take id instead of whole data. It should be generic it need to handle also all data types from the MP.

Reply:
I have adjusted my code to use the function that is already present in batch.py. I will make the necessary changes to the logic here and in batch.py in the next ticket.